src/cpp/rtps/builtin/discovery/participant/PDPServer.cpp

```diff
@@ -1224,6 +1236,7 @@ fastdds::rtps::ddb::DiscoveryDataBase& PDPServer::discovery_db()
 
 const RemoteServerList_t& PDPServer::servers()
 {
+    std::lock_guard<std::recursive_mutex> lock(*getMutex());
```

Comment:
This does not protect anything as the lock is released on returning, and the return is a const ref, which merely means that the caller cannot modify the object, but it does not prevent other threads from doing so.

In any case, this method is not used in Fast DDS or Discovery Server not even once. Although we cannot just remove public API, we can add some doxygen documentation in the header clearly stating that the returned `RemoteServerList_t&` may be modified by another thread. What do you think?

Reply:
You are right. The method is not being called within Fast DDS. I agree that we can add doxygen documentation stating that the access to the returned list may be unsafe if the remote servers list is modified at the same time by the user.